src/modules/compare/process.cpp

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2023 Team Dissolve and contributors
+
+#include "math/interpolator.h"
+#include "modules/compare/compare.h"
+
+Module::ExecutionResult CompareModule::process(ModuleContext &moduleContext)
+{
+
+    Messenger::print("COMPAREE");
+
+    // Mapping from data pair to ranges and error
+    std::map<DataSourceKeywordBase::DataPair *, rangeErrorPair> dataSourcesErrors;
+
+    int index;
+    for (auto &dataPair : dataSources_)
```

Comment:
Wouldn't it be nicer to do this here?
```suggestion
    for (auto &[dataSourceA, dataSourceB] : dataSources_)
```

Reply:
I need the pointer to the `dataPair` as they are used as the key in the `dataSourcesErrors` map:
https://github.com/disorderedmaterials/dissolve/blob/00bd84c54c2a2223f8c26f563f9c4729b6311769/src/modules/compare/process.cpp#L26
https://github.com/disorderedmaterials/dissolve/blob/00bd84c54c2a2223f8c26f563f9c4729b6311769/src/modules/compare/process.cpp#L51
https://github.com/disorderedmaterials/dissolve/blob/00bd84c54c2a2223f8c26f563f9c4729b6311769/src/modules/compare/process.cpp#L54